extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php

```diff
@@ -0,0 +1,243 @@
+				->getData(),
+			'recentlyJoined' => $this->sendRequest( 'CommunityPageSpecialController', 'getRecentlyJoinedData' )
+				->getData(),
+			'recentActivityModule' => $this->getRecentActivityData(),
+		] );
+	}
+
+	public function header() {
+		$isMember = CommunityPageSpecialHelper::userHasEdited( $this->wg->User );
+
+		$this->response->setValues( [
+			'inviteFriendsText' => $this->msg( 'communitypage-invite-friends' )->plain(),
+			'headerWelcomeMsg' => $this->msg( 'communitypage-tasks-header-welcome' )->plain(),
+			'pageListEditText' => $this->msg( 'communitypage-page-list-edit' )->plain(),
+			'thisMonthText' => $this->msg( 'communitypage-this-month' )->plain(),
+			'showMonthlySummary' => $isMember,
+			'showAdminsSummary' => !$isMember,
+			'statPagesTitle' => $this->msg( 'communitypage-pages' )->plain(),
+			'statPagesNumber' => 45,
+			'statPageViewsTitle' => $this->msg( 'communitypage-pageviews' )->plain(),
+			'statPageViewsNumber' => '1,049',
+			'statEditsTitle' => $this->msg( 'communitypage-edits' )->plain(),
+			'statEditsNumber' => 621,
+			'statEditorsTitle' => $this->msg( 'communitypage-editors' )->plain(),
+			'statEditorsNumber' => 23,
+		] );
+	}
+
+	/**
+	 * Set context for contributorsModule template. Needs to be passed through the index method in order to work.
+	 * @return array
+	 */
+	public function getTopContributorsData() {
+		$userContribCount = $this->usersModel->getUserContributions( $this->wg->user );
+		$contributors = CommunityPageSpecialUsersModel::filterGlobalBots(
+				// get extra contributors so if there's global bots they can be filtered out
+				CommunityPageSpecialUsersModel::getTopContributors( 50, '1 MONTH' )
+			);
+		// get details for only 5 of the remaining contributors
+		$contributorDetails = $this->getContributorsDetails( array_slice( $contributors, 0, 5 ) );
+
+		$this->response->setData( [
+			'topContribsHeaderText' => $this->msg( 'communitypage-top-contributors-week' )->plain(),
+			'yourRankText' => $this->msg( 'communitypage-user-rank' )->plain(),
+			'userContributionsText' => $this->msg( 'communitypage-user-contributions' )
+				->numParams( $userContribCount )
+				->text(),
+			'contributors' => $contributorDetails,
+			'userAvatar' => AvatarService::renderAvatar(
+				$this->wg->user->getName(),
+				AvatarService::AVATAR_SIZE_SMALL_PLUS
+			),
+			'userRank' => 302,
+			'memberCount' => 309,
+			'userContribCount' => $userContribCount
+		] );
+	}
+
+	/**
+	 * Set context for topAdmins template. Needs to be passed through the index method in order to work.
+	 * @return array
+	 */
+	public function getTopAdminsData() {
+		$topAdmins = CommunityPageSpecialUsersModel::filterGlobalBots(
+			// get all admins who have contributed in the last two years ordered by contributions
+			CommunityPageSpecialUsersModel::getTopContributors( 10, '2 YEAR', true )
+		);
+		$topAdminsDetails = $this->getContributorsDetails( $topAdmins );
+
+		$remainingAdminCount = count ( $topAdmins ) - 2;
```

Comment:
Minor coding convention issue: Shouldn't be a space after `count` (`count( $topAdmins )`). Same elsewhere in the changes.